### PR TITLE
Add basic buttons

### PR DESCRIPTION
**Pull request by viktor-yakubiv**

### CODE REVIEW

**Comment by viktor-yakubiv** [comment]
@Joozty this does not work for this moment. I have to fix pseudo classes.

Also, I need a parameter that describes a type of [button](https://material.io/components/buttons/), that can be either text, outlined or contained. I cannot use `type` prop because it can be passed to a `<button>` HTML element. `color` does not fit current case, `kind` I just do not like. Any suggestions?

**Comment by Joozty** [comment]
@viktor-yakubiv What about `variant`?

File: src/elements/button/button.css [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  @extend %button;
+}
+
+.button {
+  @extend %button;
+}
+
+/* Variants */
+
+.text {
```
Comment (Joozty) [inline src/elements/button/button.css]: This empty class makes me think about whether it's necessary here. I guess it's good for keeping consistency.

File: postcss.config.js [hunk 2]
```diff
@@ -13,14 +13,19 @@ module.exports = {
     calc(),
     presetEnv({
       stage: 0,
+      features: {
+        'color-mod-function': {
+          unresolved: 'ignore',
```
Comment (Joozty) [inline postcss.config.js]: Is it good to ignore it?

File: src/elements/button/button.css [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    &:focus {
+      box-shadow: var(--button-pressed-shadow), var(--button-focus-ring);
+    }
+  }
+}
+
+button,
+[type='button'],
+[type='submit'],
+[type='reset'] {
```
Comment (Joozty) [inline src/elements/button/button.css]: Interesting didn't know about this type.

File: src/elements/button/button.css [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  --button-background: var(--primary);
+  --button-color: var(--white);
+  box-shadow: 0 1px 2px #0003;
+
+  &:hover {
+    box-shadow: 0 1px 2px #0003;
+  }
+
+  &:focus {
+    box-shadow: 0 1px 2px #0003, var(--button-focus-ring);
```
Comment (Joozty) [inline src/elements/button/button.css]: Consider creating a variable for this color.

File: postcss.config.js (hunk 2 above)
Comment (viktor-yakubiv) [reply postcss.config.js]: I do not know but I used this function only once for this moment.

Comment (Joozty) [reply postcss.config.js]: I would prefer `warn` level at least.

File: src/elements/button/button.css (hunk 4 above)
Comment (viktor-yakubiv) [reply src/elements/button/button.css]: I think more about having shadow variables. Colours are also good but shadows are more important. Let's think about it later.

**Pull request merged**